src/Modules/OrchardCore.Commerce/Controllers/UserController.cs

```diff
@@ -86,12 +86,53 @@ public async Task<IActionResult> AddressesPost()
         return RedirectToAction(nameof(Addresses));
     }
 
-    private async Task<ContentItem> GetUserAddressesAsync(User user)
+    [HttpGet]
+    public async Task<IActionResult> Details()
+    {
+        if (User.Identity?.IsAuthenticated != true) return LocalRedirect("~/Login?ReturnUrl=~/user/details");
+        if (await _userManager.GetUserAsync(User) is not User user) return NotFound();
+
+        var userDetails = await GetUserContentItemAsync(user, UserDetails);
+        var editor = await _contentItemDisplayManager.BuildEditorAsync(
+            userDetails,
+            _updateModelAccessor.ModelUpdater,
+            isNew: false);
+
+        return View(editor);
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    [Route("user/details")]
+    public async Task<IActionResult> DetailsPost()
+    {
+        if (User.Identity?.IsAuthenticated != true) return LocalRedirect("~/Login?ReturnUrl=~/user/details");
+        if (await _userManager.GetUserAsync(User) is not User user) return NotFound();
+
+        var userDetails = await GetUserContentItemAsync(user, UserDetails);
+        await _contentItemDisplayManager.UpdateEditorAsync(userDetails, _updateModelAccessor.ModelUpdater, isNew: false);
+
+        if (_updateModelAccessor.ModelUpdater.ModelState.IsValid)
+        {
+            user.Put(UserDetails, userDetails);
+            _session.Save(user);
+            await _notifier.SuccessAsync(H["Your details have been updated."]);
+        }
+        else
+        {
+            var errors = _updateModelAccessor.ModelUpdater.GetModelErrorMessages();
+            foreach (var error in errors.WhereNot(string.IsNullOrEmpty)) await _notifier.ErrorAsync(H[error]);
+        }
+
+        return RedirectToAction(nameof(Details));
+    }
+
+    private async Task<ContentItem> GetUserContentItemAsync(User user, string contentType)
```

Comment:
You can user `IEntity user` in parameters instead of the exact type `User`

Reply:
What'd be the advantage of that though? This is only ever used in this controller where it's only used with the `User` type.

Reply:
Just the convention to use interfaces for parameters rather than exact classes but if it's only used and will be used at one place we can go with `User˙user`

Reply:
afaik that convention applies to public/protected, but not private. For example you can see the similar analyzer rule regarding collections: https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0016.md

Reply:
Well gathering information is always beneficial. Now I know this aswell.